src/main/java/backend/RepoID.java

```diff
@@ -0,0 +1,68 @@
+package backend;
+
+import util.Utility;
+
+import java.util.Optional;
+
+/**
+ * Represents the ID of a repository. To be used to ensure safe comparison
+ */
+public final class RepoID {
+
+    private final String repoIDString;
+    private final String repoOwner;
+    private final String repoName;
+
+    /**
+     * Return an Optional with a new instance of repoID if repoIDString is valid
+     * @param repoIDString
+     * @return Optional<RepoID>
+     */
+    public static Optional<RepoID> createRepoIDInstance(String repoIDString){
+        String lowerCaseRepoIDString = repoIDString.toLowerCase();
+        boolean isValidRepoIDString = Utility.isWellFormedRepoId(repoIDString);
```

Comment:
Maybe we could move this utility method in here? This class can handle everything related to repo ids.

Also, it might be useful to have a method for parsing repo ids, and define isWellFormedRepoId in terms of that. More DRY.